src/types/path.ts

```diff
@@ -8,11 +8,11 @@ type ExtractParamsFromPathString<
   TPath extends string,
   TParams extends Record<string, Param | undefined> = Record<never, never>
 > = TPath extends `${string}${ParamStart}${infer Param}${ParamEnd}${infer Rest}`
-  ? MergeParams<{ [P in ExtractParamName<Param>]: ExtractPathParamType<Param, TParams> }, ExtractParamsFromPathString<Rest, TParams>>
+  ? MergeParams<{ [P in Param]: ExtractPathParamType<Param, TParams> }, ExtractParamsFromPathString<Rest, TParams>>
   : Record<never, never>
 
 export type PathParams<TPath extends string> = {
-  [K in keyof ExtractParamsFromPathString<TPath>]?: Param
+  [K in ExtractParamName<keyof ExtractParamsFromPathString<TPath>>]?: Param
```

Comment:
this is what drives the 2nd argument for `path`, should remove `?` from param keys